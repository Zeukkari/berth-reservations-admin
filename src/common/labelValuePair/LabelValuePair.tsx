import React from 'react';
import { FormattedMessage } from 'react-intl';

import styles from './labelValuePair.module.scss';

interface Props {
  label: string;
  value?: string | null;
}

const LabelValuePair = ({ label, value = '-' }: Props) => (
  <div>
    <FormattedMessage id={label}>
      {(lbl: string) => <span className={styles.label}>{lbl}</span>}
    </FormattedMessage>
    <span className={styles.label}>:</span>
    <span className={styles.labelValue}>{value}</span>
  </div>
);

export default LabelValuePair;
