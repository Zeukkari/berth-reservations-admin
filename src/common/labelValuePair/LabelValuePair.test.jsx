import { shallow } from 'enzyme';
import React from 'react';
import { FormattedMessage } from 'react-intl';

import LabelValuePair from './LabelValuePair';

const dummyProps = { label: 'dummyLabel', value: 'bar' };

describe('LabelValuePair', () => {
  const getWrapper = () => shallow(<LabelValuePair {...dummyProps} />);

  test('should render a wrapper with a className of "vene-label-value-pair"', () => {
    expect(getWrapper().find('div.vene-label-value-pair')).toHaveLength(1);
  });

  test('should pass the label to FormattedMessage', () => {
    const label = 'dummyLabel';
    const value = 'bar';

    expect(
      getWrapper({ label })
        .find(FormattedMessage)
        .prop('id')
    ).toBe(label);
  });
});
