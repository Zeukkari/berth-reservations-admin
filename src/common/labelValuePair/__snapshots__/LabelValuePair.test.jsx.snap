// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelValuePair renders normally 1`] = `"<div><span class=\\"label\\">foo:</span><span class=\\"labelValue\\">bar</span></div>"`;
